tests/snuba/search/test_backend.py

```diff
@@ -1973,6 +1973,42 @@ def test_message_negation(self):
 
         assert list(results) == list(results2)
 
+    def test_error_main_thread_true(self):
+        self.store_event(
+            data={
+                "fingerprint": ["put-me-in-group1"],
+                "event_id": "2" * 32,
+                "message": "something",
+                "timestamp": iso_format(self.base_datetime),
+                "exception": {
+                    "values": [
+                        {
+                            "type": "SyntaxError",
+                            "value": "hello world",
+                            "thread_id": 1,
+                        },
+                    ],
+                },
+                "threads": {
+                    "values": [
+                        {
+                            "id": 1,
+                            "main": True,
+                        },
+                    ],
+                },
+            },
+            project_id=self.project.id,
+        )
+
+        results = self.make_query(
+            projects=[self.project],
+            search_filter_query="error.handled:1",
+            sort_by="date",
+        )
+
+        assert len(results) == 1
```

Comment:
@barkbarkimashark ideas why the result is 2 instead of 1?
It should be 1 only, the same if I do `results.hits == 1`.

Reply:
Try create a new project and storing the event associated with the newly created project. The test `setUp()` is associating events and such with `self.project` which is also returning in your query.

Reply:
Good point, thanks.